server/eclipse-project/src/main/java/us/freeandfair/corla/model/Administrator.java

```diff
@@ -87,16 +88,14 @@
   private String my_full_name;
   
   /**
-   * The two-factor authentication information.
+   * The last login time.
    */
-  // TODO this is a placeholder, the final format of this is not yet known
-  @Column(nullable = false, updatable = false)
-  private String my_two_factor_auth_info;
+  private Instant my_last_login_time;
   
   /**
-   * The last login time.
+   * The last logout time.
    */
-  private Instant my_last_login_time;
+  private Instant my_last_logout_time;
```

Comment:
Ah... is it to update the last logout time that you want a username submitted to the logout/deauthenticate endpoint? I still think it's unnecessary - you get the actual administrator object from the session, and can just set it then.

Reply:
Yes, that is the purpose, though I was still on the fence about its utility.  You'll see the design didn't migrate over into `AbstractAuthentication`.

Reply:
I got rid of it, but `deauthenticate` does still pull the username from the active session if there is one when it calls the child classes.